src/Reflection/InitializerExprTypeResolver.php

```diff
@@ -1354,7 +1354,7 @@ public function resolveEqualType(Type $leftType, Type $rightType): BooleanType
 			return $this->resolveConstantArrayTypeComparison($leftType, $rightType, fn ($leftValueType, $rightValueType): BooleanType => $this->resolveEqualType($leftValueType, $rightValueType));
 		}
 
-		return new BooleanType();
+		return $leftType->looseCompare($rightType);
```

Comment:
This whole method needs to be just `$leftType->looseCompare($rightType)`.

Reply:
In case we want to move all the logic we have here into `$leftType->looseCompare($rightType)` we would need to pass the PhpVersion was an additional param, right?
(Loose comparison differs between php versions)